sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -385,14 +385,21 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
         val (relationCtx, cols, partition, ifPartitionNotExists, byName)
         = visitInsertIntoTable(table)
         withIdentClause(relationCtx, ident => {
-          InsertIntoStatement(
+          val insertIntoStatement = InsertIntoStatement(
             createUnresolvedRelation(relationCtx, ident),
             partition,
             cols,
             query,
             overwrite = false,
             ifPartitionNotExists,
             byName)
+          val optimizeInsertIntoValues = conf.getConf(
```

Comment:
optional: I think you can just inline this `conf.getConf(...)` call into the single call site below on L398